src/main/drivers/timer.h

```diff
@@ -104,10 +104,14 @@ typedef struct timerHardware_s {
     DMA_Channel_TypeDef *dmaRef;
 #endif
     uint8_t dmaIrqHandler;
-#if defined(STM32F4) || defined(STM32F7)
+#if defined(STM32F3) || defined(STM32F4) || defined(STM32F7)
```

Comment:
This should probably be simplified to `#if !defined(STM32F1)`.

Reply:
I wonder why it isn't `#ifdef USE_DSHOT_DMAR` as TIM_UP DMA is not used otherwise.

Reply:
Not sure if adding a dependency on the internals of other drivers into `timer.h` is the right thing to do, since this is the foundation that a lot of other drivers rely upon. Having this as a pure hardware dependend definition is probably better.

Reply:
Agreed, I thought there was another usage of this define above but turns out it was my imagination.

Reply:
@DieHertz: Are you going to change this, or is this good to be merged?